sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -228,8 +228,11 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case DescribeColumn(_: ResolvedTable, _, _) =>
       throw new AnalysisException("Describing columns is not supported for v2 tables.")
 
-    case DropTable(catalog, ident, ifExists) =>
-      DropTableExec(catalog, ident, ifExists) :: Nil
+    case DropTable(r: ResolvedTable, ifExists, _) =>
```

Comment:
not related to this PR, but we should think about how to handle the `purge` flag in v2 API.

Reply:
The `purge` option seems that it should be handled by the catalog implementation, not by Spark (e.g., `ifExists`). We can update `TableCatalog.dropTable` to pass an additional flag, or we could introduce `TableCatalog.purgeTable` for the catalog implementor can optionally implement.

Reply:
How useful is the `purge` option? If it's rarely used, we can just fail v2 DROP TABLE if `purge` is specified.

cc @rdblue @brkyvz

Reply:
I think it would make sense to add a variant of `dropTable` that accepts a purge option. Then have a default implementation that ignores it and calls `dropTable` without it. This seems like something that catalogs should be passed.

Reply:
Silently ignore the `purge` option as the default behavior looks risky. How about making the new variant of `dropTable` fail by default if `purge==true`?

Reply:
I created #30267 based on the feedback.